pkg/apis/topology/v1alpha1/types.go

```diff
@@ -29,24 +29,32 @@ type NodeResourceTopology struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	TopologyPolicy string        `json:"topologyPolicy"`
-	Nodes   []NUMANodeResource   `json:"nodes"`
+	TopologyPolicies []string        `json:"topologyPolicies"`
+	Zones            map[string]Zone `json:"zones"`
 }
 
-// NUMANodeResource is the spec for a NodeResourceTopology resource
-type NUMANodeResource struct {
-	NUMAID int
-	Resources v1.ResourceList
+// Zone is the spec for a NodeResourceTopology resource
+type Zone struct {
+	Type       string
+	Parent     string
+	Costs      map[int]int
+	Attributes map[string]int
+	Resources  ResourceInfoMap
 }
 
+type ResourceInfo struct {
+	Allocatable int
```

Comment:
Shouldn't allocatable and capacity be of the type v1.ResourceList ?

Reply:
Ah, Now I understand why you have an integer here! Nevermind :)

Reply:
yes, it's more practical.